src/integrationtests/java/com/ericsson/ei/artifactintegrationtests/ArtifactFlowStepsIT.java

```diff
@@ -0,0 +1,167 @@
+                .performRequest();
+        assertEquals(200, response.getStatusCodeValue());
+    }
+
+
+    @When("^eiffel events are sent$")
+    public void eiffel_events_are_sent() throws Throwable  {
+        ObjectMapper objectMapper = new ObjectMapper();
+
+        final URL upStreamInput = new File(UPSTREAM_INPUT_FILE).toURI().toURL();
+        ArrayNode upstreamJson = (ArrayNode) objectMapper.readTree(upStreamInput);
+        if (upstreamJson != null) {
+            for (JsonNode event : upstreamJson) {
+                String eventStr = event.toString();
+                rabbitMqTemplate.convertAndSend(eventStr);
+            }
+        }
+
+        super.sendEventsAndConfirm();
+    }
+
+    @Then("^mongodb should contain mail\\.$")
+    public void mongodb_should_contain_mails() throws Throwable {
+        JsonNode newestMailJson = getMailFromDatabase();
+        String createdDate = newestMailJson.get("created").get("$date").asText();
+
+        long createdDateInMillis = ZonedDateTime.parse(createdDate).toInstant().toEpochMilli();
+        assert(createdDateInMillis >= startTime): "Mail was not triggered. createdDateInMillis is less than startTime.";
+    }
+
+    @Override
+    protected String getRulesFilePath() {
+        return RULES_FILE_PATH;
+    }
+
+    @Override
+    protected String getEventsFilePath() {
+        return EVENTS_FILE_PATH;
+    }
+
+    @Override
+    protected int extraEventsCount() {
+        return 8;
+    }
+
+    @Override
+    protected List<String> getEventNamesToSend() {
+        List<String> eventNames = new ArrayList<>();
+        eventNames.add("event_EiffelConfidenceLevelModifiedEvent_3_2");
+        eventNames.add("event_EiffelArtifactPublishedEvent_3");
+        eventNames.add("event_EiffelArtifactCreatedEvent_3");
+        eventNames.add("event_EiffelTestCaseTriggeredEvent_3");
+        eventNames.add("event_EiffelTestCaseStartedEvent_3");
+        eventNames.add("event_EiffelTestCaseFinishedEvent_3");
+        eventNames.add("event_EiffelArtifactPublishedEvent_3_1");
+        eventNames.add("event_EiffelConfidenceLevelModifiedEvent_3");
+        eventNames.add("event_EiffelTestCaseTriggeredEvent_3_1");
+        eventNames.add("event_EiffelTestCaseStartedEvent_3_1");
+        eventNames.add("event_EiffelTestCaseFinishedEvent_3_1");
+        return eventNames;
+    }
+
+    @Override
+    protected Map<String, JsonNode> getCheckData() throws IOException {
+        JsonNode expectedJSON = getJSONFromFile(AGGREGATED_OBJECT_FILE_PATH);
+        Map<String, JsonNode> checkData = new HashMap<>();
+        checkData.put(AGGREGATED_OBJECT_ID, expectedJSON);
+        return checkData;
+    }
+
+    private JsonNode getMailFromDatabase() throws IOException {
+        ArrayList<String> allMails = mongoDBHandler.getAllDocuments(MAILHOG_DATABASE_NAME, "messages");
+        String mailString = allMails.get(0);
+
+        ObjectMapper mapper = new ObjectMapper();
```

Comment:
You could maybe move "ObjectMapper mapper = new ObjectMapper()" out, you are doing this in many functions.